OpenOversight/app/utils.py

```diff
@@ -226,3 +226,11 @@ def compute_leaderboard_stats(select_top=25):
                             .order_by(func.count(Face.user_id).desc()) \
                             .limit(select_top).all()
     return top_sorters, top_taggers
+
+
+def badge_exists(star_no, department_id):
+    """badge_exists checks if a given badge number exists already for a department."""
+    for assignment in Assignment.query.filter_by(star_no=star_no).all():
+        if str(assignment.baseofficer.department_id) == str(department_id):
+            return True
+        return False
```

Comment:
So here's a tricky thing about some police departments we have in OpenOversight: sometimes a badge number may appear several times - e.g. for Chicago Police Department, they allocate an officer a new badge number upon promotion and then re-use the old badge number for a different officer as it has been freed. Since we track the assignments history of officers, the same badge number might appear multiple times in the database if it was indeed allocated to another officer (yep this is silly but it happens 🤷‍♀️).

A more robust approach would be to have this function be `officer_exists` and check that no row exists with the same badge number, first name, or last name. 

Also apologies for the long overdue review. I am happy to take over this PR if you have other stuff you want to work on now 😇 :heart: